src/resource/archive/O2rArchive.cpp

```diff
@@ -111,23 +111,12 @@ bool O2rArchive::WriteFile(const std::string& filename, const std::vector<uint8_
     }
 
     // Add or replace the file in the ZIP archive
-    zip_int64_t index = zip_name_locate(mZipArchive, filename.c_str(), 0);
-    if (index >= 0) {
-        // File exists, replace it
-        if (zip_file_replace(mZipArchive, index, source, ZIP_FL_OVERWRITE) < 0) {
-            SPDLOG_ERROR("Failed to replace file \"{}\" in ZIP", filename);
-            zip_source_free(source);
-            return false;
-        }
-    } else {
-        // File doesn't exist, add it
-        if (zip_file_add(mZipArchive, filename.c_str(), source, ZIP_FL_ENC_UTF_8) < 0) {
-            SPDLOG_ERROR("Failed to add file \"{}\" to ZIP", filename);
-            zip_source_free(source);
-            return false;
-        }
+    if (zip_file_add(mZipArchive, filename.c_str(), source, ZIP_FL_ENC_UTF_8 | ZIP_FL_OVERWRITE) < 0) {
+        SPDLOG_ERROR("Failed to add file \"{}\" to ZIP", filename);
+        zip_source_free(source);
+        return false;
     }
-    printf("Success wrote file\n");
+    SPDLOG_INFO("Success wrote file");
 
     // Save changes to disk
     if (zip_close(mZipArchive) < 0) {
```

Comment:
I have someone hitting this a lot on Windows, would be nice if error message had more info

Reply:
Which game is already making use of the writefile method? This function is new of like, last week?

Reply:
Delete the first printf line

replace the second one with
SPDLOG_INFO("Successfully wrote file: {}", filename.c_str());

Reply:
It's also worth mentioning, if other ports are using WriteFile note that it is not supported in the OTR format, only O2R.

Reply:
I actually thought I did call `ResetVirtualFileSystem` but I probably need to add that.

Reply:
Finally, you're correct that the WriteFile function in ArchiveManager does not return the correct boolean if the function fails. I'll PR a fix for that.

Reply:
https://github.com/HarbourMasters/Shipwright/pull/5435 I'm using this API while trying to update SOH's accessibility build to latest

Reset isn't needed since the workflow is `turn on extraction` -> `reset soh` -> `extraction runs` -> `reset soh` -> `it works`

It works for me on Linux, generating 250MB o2r. & it seemed to work for Malkierian on Windows. But for others they're failing to write many of the files & end up with an incomplete archive that's a tenth the size it should be

Reply:
re archive manager return, that's not what I'm talking about

https://libzip.org/documentation/zip_close.html

> If successful, archive is freed. Otherwise archive is left unchanged and must still be freed.

When this function returns false the caller isn't able to tell whether that's from `zip_close` failing or `zip_open` failing

Reply:
not requirement. this design choice was made by original creators of this experiment, I'd like to move away from the current design

relevant discussion: https://github.com/HarbourMasters/Shipwright/pull/3207#discussion_r1328327428